storage-operator/deploy/role.yaml

```diff
@@ -4,36 +4,63 @@ metadata:
   creationTimestamp: null
   name: storage-operator
 rules:
+# For recording transition events
 - apiGroups:
   - ""
   resources:
-  - pods
-  - nodes
-  - services
-  - endpoints
-  - persistentvolumes
-  - persistentvolumeclaims
   - events
-  - configmaps
-  - secrets
   verbs:
-  - '*'
+  - create
+  - patch
+# For setting up monitoring for itself
+- apiGroups:
+  - ""
+  resources:
+  - services
+  verbs:
+  - get
+  - create
+  - update
 - apiGroups:
   - apps
   resources:
   - deployments
-  - daemonsets
   - replicasets
-  - statefulsets
   verbs:
-  - '*'
+  - get
```

Comment:
Why does the storage-operator need to read deployments and replicasets?

Reply:
It comes from `kubemetrics`: when creating the Service, it reads its Pod's owner controller (a ReplicaSet) and this controller's controller (a Deployment) to determine the right `selector` to use, I think. The information is disseminated in several issues and pull requests in `operator-sdk`, and I found most of them when investigating the `Failed to list *unstructured.Unstructured` error logs. Could do some digging up my history if you want, but I can confirm that there were unauthorized errors if I remove those permissions.